source/mqtt/Mqtt5Client.cpp

```diff
@@ -516,19 +604,40 @@ namespace Aws
                 return result == AWS_OP_SUCCESS;
             }
 
+            void Mqtt5Client::Close() noexcept
+            {
+                aws_rw_lock_wlock(&m_client_lock);
+                if (m_client != nullptr)
+                {
+                    aws_mqtt5_client_release(m_client);
+                    /* Waiting for client termination */
+                    std::unique_lock<std::mutex> lock(m_terminationMutex);
+                    m_terminationCondition.wait(lock, [this] { return m_terminationPredicate == true; });
```

Comment:
This is a deadlock if someone calls Close() from an event loop triggered callback.

Why do we need to wait anyways?